src/ui/controls/edit.cpp

```diff
@@ -2120,11 +2157,22 @@ void CEdit::SetMultiFont(bool bMulti)
     m_format.clear();
 
     if (bMulti)
-    {
-        m_format.reserve(m_maxChar+1);
-        for (int i = 0; i <= m_maxChar+1; i++)
-        {
-            m_format.push_back(m_fontType);
+    {
+        if ( !IsInfinite() )
```

Comment:
It's not really good to depend on the value of IsInfinite() in code like this, using std::string like mentioned in other comments should help eliminate that.